Zoo/Tour/TourOrTwoToursParallel.cs

```diff
@@ -0,0 +1,147 @@
+﻿using System.Collections;
+using Zoo.EventLogger;
+using Zoo.Util.Enum;
+using Zoo.Util.CustomException;
+
+namespace Zoo.Tour
+{
+    /// <summary>
+    /// class represents one tour or two tours at the same time
+    /// </summary>
+    internal sealed class TourOrTwoToursParallel
+    {
+        public Action touring;
+
+        public Visitor[] visitor = new Visitor[5];
+
+        public List<TourPlace> places;
+
+        private readonly List<List<TourPlace>> legallTourPlaces = new()
+        {
+            new List <TourPlace> ()
+            {
+                new TourPlace(ZonesType.OrdinaryLandZone),
+                new TourPlace(ZonesType.AmphibiaZone)
+            },
+
+            new List<TourPlace>
+            {
+                new TourPlace(ZonesType.OrdinaryLandZone),
+                new TourPlace(ZonesType.SeaCreatureZone)
+            },
+
+            new List<TourPlace>
+            {
+                new TourPlace(ZonesType.AmphibiaZone),
+                new TourPlace(ZonesType.SeaCreatureZone)
+            }
+        };
+
+        private void CheckIfTourCanBeParallel(List<TourPlace> places)
+        {
+            try
+            {
+                if (places.Count == 0)
+                {
+                    throw new NoTourWasGivenException("No tour was given");
+                }
+
+                if (places.Count >= 3)
+                {
+                    throw new CannotBeTourInThreeOrMorePlacesAtTheSameTimeException("Make several tours at the same time");
+                }
+
+                if (places.Count == 1)
+                {
+                    return;
+                }
+
+                TourPlace tourPlace1 = places[0];
+                TourPlace tourPlacetourPlace2 = places[1];
+
+                if (tourPlace1.ZooZonesTypeZooZonesType == tourPlacetourPlace2.ZooZonesTypeZooZonesType)
+                {
+                    throw new TwoOfTheSameTourException($"There are two of {tourPlace1.ZooZonesTypeZooZonesType}");
+                }
+
+                bool isLegalCombination = false;
+
+
+
+                foreach (List<TourPlace> legalCombination in legallTourPlaces)
+                {
+                    BitArray bitArray = new(2);
+
+                    bitArray[0] = legalCombination[0].IsEqual(tourPlace1) && legalCombination[1].IsEqual(tourPlacetourPlace2);
+                    bitArray[1] = legalCombination[0].IsEqual(tourPlacetourPlace2) && legalCombination[1].IsEqual(tourPlace1);
```

Comment:
Pay attention that you do the same thing twice with a different variable. That's what functions are for.